utilities/src/main/java/org/craftercms/commons/aws/AwsUtils.java

```diff
@@ -36,95 +34,51 @@ public final class AwsUtils {
                 .build();
     }
 
     /**
      * Copy a list of objects from a source bucket to a target bucket
      *
-     * @param client          The {@link AmazonS3} client
-     * @param executorFactory The {@link ExecutorFactory} to use to create a {@link TransferManager}
+     * @param client          The {@link S3AsyncClient} client
      * @param sourceBucket    The source bucket
      * @param sourceBaseKey   The base key in the source bucket (i.e. prefix for all paths)
      * @param targetBucket    The target bucket
      * @param targetBaseKey   The base key in the target bucket (i.e. prefix for all paths)
      * @param paths           The list of paths to copy
-     * @throws InterruptedException If the operation is interrupted while copying the objects
      */
-    public static void copyObjects(AmazonS3 client, ExecutorFactory executorFactory, String sourceBucket, String sourceBaseKey,
-                                   String targetBucket, String targetBaseKey, List<String> paths) throws InterruptedException {
+    public static void copyObjects(S3AsyncClient client, String sourceBucket, String sourceBaseKey,
+                                   String targetBucket, String targetBaseKey, List<String> paths) {
 
-        TransferManager transferManager = buildTransferManager(client, executorFactory);
+        S3TransferManager transferManager = buildTransferManager(client);
         try {
-            CountDownLatch doneSignal = new CountDownLatch(paths.size());
             logger.debug("Copying {} objects from '{}' to '{}'", paths.size(), sourceBucket, targetBucket);
             for (String path : paths) {
                 logger.debug("Copying '{}' from '{}' to '{}'", path, sourceBucket, targetBucket);
                 String sourceKey = sourceBaseKey + path;
                 String targetKey = targetBaseKey + path;
-                CopyObjectRequest copyRequest = new CopyObjectRequest()
-                        .withSourceBucketName(sourceBucket)
-                        .withSourceKey(sourceKey)
-                        .withDestinationBucketName(targetBucket)
-                        .withDestinationKey(targetKey);
-                transferManager.copy(copyRequest, new MultiOperationTransferStateChangeListener(sourceBucket, sourceKey, targetBucket, targetKey, doneSignal));
+                CopyObjectRequest copyObjectRequest = CopyObjectRequest.builder()
+                        .sourceBucket(sourceBucket)
+                        .sourceKey(sourceKey)
+                        .destinationBucket(targetBucket)
+                        .destinationKey(targetKey)
+                        .build();
+                CopyRequest copyRequest = CopyRequest.builder()
+                        .copyObjectRequest(copyObjectRequest)
+                        .build();
+                Copy copy = transferManager.copy(copyRequest);
+                CompletedCopy completedCopy = copy.completionFuture().join();
```

Comment:
Won't this `join` make the copy happen one by one sequentially?

Reply:
I updated this to use the executor.